controllers/managedocs_controller.go

```diff
@@ -146,3 +156,32 @@ func (r *ManagedOCSReconciler) setDesiredStorageCluster(
 
 	return nil
 }
+
+func (r *ManagedOCSReconciler) updateComponents(req ctrl.Request) (v1.ManagedOCSComponents, error) {
+	var components v1.ManagedOCSComponents
+
+	// Checking the StorageCluster component.
+	var sc ocsv1.StorageCluster
+
+	scNamespaceName := types.NamespacedName{
+		Name:      storageClusterName,
+		Namespace: req.Namespace,
+	}
+
+	err := r.Get(r.ctx, scNamespaceName, &sc)
+	if err == nil {
+		if sc.Status.Phase == "Ready" {
+			components.StorageCluster.Status = v1.ComponentReady
+		} else {
+			components.StorageCluster.Status = v1.ComponentPending
+		}
+	} else if errors.IsNotFound(err) {
+		components.StorageCluster.Status = v1.ComponentNone
+	} else {
+		r.Log.Error(err, "error getting StorageCluster")
+		return components, err
+	}
+	components.StorageCluster.LastChecked = metav1.Now()
```

Comment:
`metav1.Now()` should only be used if we cannot infere the last update time from the actual resource. 
Please check if we can for `stroagecluster` before falling back to `Now()`

Reply:
How is this field going to be used?

Reply:
I'm currently using this field to track the last time we checked on this component.

Reply:
what is the value in infering the last update time of the actual resource, assuming it even reports it.

Reply:
We should give the most accurate value we can give, the check time should be a fall back only if we cannot get a more accurate time somehow

Reply:
The value is in aggregating and reporting the most accurate time to the entity which consumes this information.  
Think about a case where the storagecluster is not being reconciled by the OCS operator at all. The time should be fixed but If you take the sampling time it will change with each reconcile. 

Now In some cases, we do not have a source to read an actual update time from, in that case, the sample time is the best approximation we have.

Reply:
storagecluster has no time field.

Reply:
Ok, what about `ManagedFields` in the metadata section?
they are not mandatory, but if exists we can use it

Reply:
there is nothing to indicate that those times are updated by the storagecluster being updated.

Reply:
Look at the definition for ManagedField

Reply:
272   // ManagedFields maps workflow-id and version to the set of fields                                                                         
 273   // that are managed by that workflow. This is mostly for internal                                                                          
 274   // housekeeping, and users typically shouldn't need to set or                                                                              
 275   // understand this field. A workflow can be the user's name, a                                                                             
 276   // controller's name, or the name of a specific apply path like                                                                            
 277   // "ci-cd". The set of fields is always in the version that the                                                                            
 278   // workflow used when modifying the object.                                                                                                
 279   //                                                                                                                                         
 280   // +optional                                                                                                                               
 281   ManagedFields []ManagedFieldsEntry `json:"managedFields,omitempty" protobuf:"bytes,17,rep,name=managedFields"`                                           

1148   // Time is timestamp of when these fields were set. It should always be empty if Operation is 'Apply'                                      
1149   // +optional                                                                                                                               
1150   Time *Time `json:"time,omitempty" protobuf:"bytes,4,opt,name=time"`

Reply:
?

Reply:
Ok, so in this case, lets go with the sample time and we can request ocs-operator in the future to provide a proper time.

Reply:
I gave another thought about this matter. If the only component we have now is StorageCluster and we have no means to get the update time then you can remove it from the struct, but keep the struct as an extension point.

In the future when we will have more components we can reevaluate